keras_cv/layers/preprocessing/random_choice.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomChoice(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """RandomChoice constructs a pipeline based on provided arguments.
+
+    The implemented policy does the following: for each inputs provided in `call`(), the
+    policy selects a random layer from the provided list of `layers`.  It then calls the
+    `layer()` on the inputs.
+
+    Usage:
+    ```python
+    # construct a list of layers
+    layers = keras_cv.layers.RandAugment.get_standard_policy(
+        value_range=(0, 255), magnitude=0.75, magnitude_stddev=0.3
+    )
+    layers = layers[:4]  # slice out some layers you don't want for whatever reason
+    layers = layers + [keras_cv.layers.GridMask()]
+
+    # create the pipeline.
+    pipeline = keras_cv.layers.RandomChoice(layers=layers)
+
+    augmented_images = pipeline(images)
+    ```
+
+    Args:
+        layers: a list of `keras.Layers`.  These are randomly inputs during
+            augmentation to augment the inputs passed in `call()`.  The layers passed
+            should subclass `BaseImageAugmentationLayer`.
+        auto_vectorize: whether to use `tf.vectorized_map` or `tf.map_fn` to
+            apply the augmentations.  This offers a significant performance boost, but
+            can only be used if all the layers provided to the `layers` argument
+            support auto vectorization.
+        seed: Integer. Used to create a random seed.
+    """
+
+    def __init__(
+        self,
+        layers,
+        auto_vectorize=False,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs, seed=seed, force_generator=True)
+        self.layers = layers
+        self.auto_vectorize = auto_vectorize
+        self.seed = seed
+
+    def _curry_call_layer(self, inputs, layer):
+        def call_layer():
+            return layer(inputs)
+
+        return call_layer
+
+    def _augment(self, inputs, *args, **kwargs):
```

Comment:
I think we actually want to do this on a sample-wise basis, not a batch-wise basis.

As of now, it looks like this does everything on a batch-wise basis.  Thoughts?

Reply:
It is on a sample-wise basis. The `_augment` method of `BaseImageAugmentationLayer` is passed single samples as inputs. It is mapped to a batch in the `_batch_augment`.

Sample-wise basis is also tested for here: https://github.com/keras-team/keras-cv/pull/434/files#diff-ba41275c34609917e7f01d01bfe4f36d015082795580f91ab30bc3b9aa4cece2R71

Reply:
Oops!  Yep, you're right.  Sorry!